include/zephyr/drivers/mfd/max31790.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2024 SILA Embedded Solutions GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_MFD_MAX31790_H_
+#define ZEPHYR_INCLUDE_DRIVERS_MFD_MAX31790_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/device.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
+
+/**
+ * @defgroup mdf_interface_max31790 MFD MAX31790 interface
+ * @ingroup mfd_interfaces
+ * @{
+ */
+
+#define MAX31790_OSCILLATOR_FREQUENCY_IN_HZ 32768
+#define MAX31790_PWMTARGETDUTYCYCLE_MAXIMUM ((1 << 9) - 1)
+#define MAX31790_TACHTARGETCOUNT_MAXIMUM    ((1 << 11) - 1)
+#define MAX31790_CHANNEL_COUNT              6
+#define MAX31790_RESET_TIMEOUT_IN_US        1000
+
+#define MAX37190_REGISTER_GLOBALCONFIGURATION               0x00
+#define MAX37190_REGISTER_PWMFREQUENCY                      0x01
+#define MAX37190_REGISTER_FANCONFIGURATION(channel)         (0x02 + channel)
+#define MAX31790_REGISTER_FANDYNAMICS(channel)              (0x08 + channel)
+#define MAX37190_REGISTER_FANFAULTSTATUS1                   0x11
+#define MAX37190_REGISTER_TACHCOUNTMSB(channel)             (0x18 + 2 * channel)
+#define MAX31790_REGISTER_PWMOUTTARGETDUTYCYCLEMSB(channel) (0x40 + 2 * channel)
+#define MAX31790_REGISTER_TACHTARGETCOUNTMSB(channel)       (0x50 + 2 * channel)
+
+#define MAX37190_GLOBALCONFIGURATION_STANDBY_BIT             BIT(7)
```

Comment:
Defines that are only used a a single C file should be moved out of this header

Reply:
I'm using some of these defines only in a single C file, but some in multiple files. Therefore I would have to spread this information over several files, but only partially. I think this would be more confusing than having them all together in one file.

Reply:
In that case could you move it to be a private header in `drivers/mfd/`?

Reply:
done

Reply:
Done, but build fails :-)

Looks like the include directory zephyr/drivers is missing on native_posix, but is available for a build on nucleo_h743zi. Did you mean to put the header into zephyr/drivers/mfd? Because then I will try to figure out the difference between the build on the two boards. Or what do you mean with "private header", maybe I just misunderstood you?

Reply:
Apologies, I got mixed up with another PR.
Moving it would only work if the header is in the same folder as the source, not share between multiple drivers.

Reply:
Yes, I got the same feeling. Otherwise I would have had to add it to the include directories, which would have made it less "private". So keep it public where it is?

Reply:
I reverted it back to the public version, as this should at least build.